### PR TITLE
BUGFIX: Change workspace runs into endless while-loop

### LINKED ISSUE #3787: BUG: 9.0 `change-base-workspace` is infinite loop and times out - always?!!
In my tests changing the base workspace does not succeed:

image

(at least im not able to switch back to live)

This might be because the code that is bugged in 8.3 https://github.com/neos/neos-ui/pull/3734 doesnt work in 9.0 either.
(at least in the code there is a `while true` ...

https://github.com/neos/neos-ui/blob/28fb9c5e8e92a88fe8d79666b454b19c5533cff3/Classes/Controller/BackendServiceController.php#L469

Scenario 1:

- create workspace
- create new subpage in workspace and publish and highlight it
- change workspace to live again

->  fails currently

But scenario 2

- create workspace
- create new subpage in workspace
- create new subsubpage in workspace
- publish and highlight the new subsubpage
- change workspace to live again

->  fails currently

### Description

[Description of the bug or feature]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

#### Expected behavior

[What you expected to happen]

#### Actual behavior

[What actually happened]

### Affected Versions

Neos:

UI:

You can get this information by running `composer show` or using the package management module within Neos.
The UI version can also be found inside the left sidebar after version 1.0.3.

### LINKED ISSUE #5129: BUG: Can't switch from internal workspace back to live
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In latest 9.0 to switch from a "internal workspace" back to "live" the process seems to run into an infinite loop. CPU goes up to 100% and no response anymore. 

### Expected Behavior

I can switch back to live workspace as baseworkspace of my user workspace.

### Steps To Reproduce

Create new internal Workspace "Test"
Switch to "Test"
Switch to "Live"

### Environment

```markdown
- Flow: 9.0
- Neos: 9.0
- PHP: 8.2
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pKallert**
**Review instructions**
Currently when switching the workspace in the UI it just runs endlessly and the docker container complains if you switch workspaces too often.  

Apparently the bug was introduced here: [PR](https://github.com/neos/neos-development-collection/commit/7d691d85394c9819b64996c51cc208f321603212#diff-3e5de1a138586c8c85cdd71ed1a765ecc0ba0d79ab2579a6000475d427cefda2L944 )

The line used to be 
`$nextBaseWorkspace = $contentRepository->getWorkspaceFinder()->findOneByName($nextBaseWorkspace->baseWorkspaceName);`

and was changed to 
`$nextBaseWorkspace = $this->requireBaseWorkspace($workspace, $workspaceFinder);`

since we always check for $workspace, the $nextBaseWorkspace always stays the same
**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php [hunk 1]
```diff
@@ -960,13 +960,12 @@ private function requireNonCircularRelationBetweenWorkspaces(Workspace $workspac
         if ($workspace->workspaceName->equals($baseWorkspace->workspaceName)) {
             throw new BaseWorkspaceEqualsWorkspaceException(sprintf('The base workspace of the target must be different from the given workspace "%s".', $workspace->workspaceName->value));
         }
-
         $nextBaseWorkspace = $baseWorkspace;
-        while ($nextBaseWorkspace->baseWorkspaceName !== null) {
+        while (!is_null($nextBaseWorkspace->baseWorkspaceName)) {
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php:964]: Just wondering: Is that related to the fix? `!is_null(x)` and `x !== null` should behave the same?!

**Review by bwaidelich — APPROVED** [review]
doh :)

**Review by kitsunet — APPROVED** [review]
Oof yeah, thanks for finding this!

**Comment by kitsunet** [comment]
I think this needs a rebase to after the DBAL changes?! That probably fixes the two CI problems.

File: Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php (hunk 1 above)
Comment (pKallert) [reply Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php:964]: Not really, I just wanted to make it consistent with requireBaseWorkspace

**Comment by mhsdesign** [comment]
Before merging we should definitely come up with a test for this. The method is used in change base workspace which is apparently untested.

**Comment by kitsunet** [comment]
All dependencies fixed, tests green, as discussed today in our meeting, we merge this before the test as it's a nasty bug.
